src/content/docs/tutorials/rust/social_network.mdx

```diff
@@ -0,0 +1,222 @@
+```Rust
+    let result = conn.query("CALL SHOW_TABLES() RETURN *")?;
+```
+Storing the result in the variable `result`. Which we can then access the content by using code snippet like this:
+```Rust
+    for row in result {
+        let row_str = row.iter()
+            .map(|col| col.to_string())  // Convert each column to a string
+            .collect::<Vec<String>>()    // Collect into a Vec<String>
+            .join(", ");                 // Join elements with ", "
+    
+        println!("{}", row_str);
+    }
+```
+Alternatively, we can also export the data into csv files for external processing.
+``` Rust
+    conn.query("COPY (CALL SHOW_TABLES() RETURN *) TO 'tables.csv' (header=true")?;
+```
+For more information on data exporting, please referring to our documentation on [Export data](../../export/index.mdx);
+
+## Importing Data
+After creating the tables, we can import the data. We do so by using the `COPY FROM` commands, like this:
+```Rust
+    conn.query("COPY User FROM './data/tutorial_user.csv'")?;
+    conn.query("COPY Post FROM './data/tutorial_posts.csv'")?;
+    conn.query("COPY FOLLOWS FROM './data/TUTORIAL_FOLLOWS.csv'")?;
+    conn.query("COPY POSTS FROM './data/TUTORIAL_POSTS.csv'")?;
+    conn.query("COPY LIKES FROM './data/TUTORIAL_LIKES.csv'")?;
+```
+For more information on loading in data, please refer to our documentation on  [import data](../../import/csv.md)
+
+## Querying queries
+We can now proceed to querying the database to answer some questions about the social network. Here are some examples of what queries can be answered with ease using Kùzu:
+
+### Q1: Which user has the most followers? And how many followers do they have?
+We can break the query down into a few steps:
+1. We can first use the `MATCH` query to find the relationship specified, in this case, we want to look at `User` node at the end of the relation `FOLLOWS`:
+```Rust
+conn.query("MATCH (u1:User)-[f:FOLLOWS]->(u2:User) RETURN u2.username")?;
+```
+For more information on `MATCH`, please refer to our documentation in [Cypher Manual](../../cypher/index.md).
+
+For each relation `FOLLOWS`, this will return the followee's username of that relation. While this doesn't provide much useful information yet, we may improve on this query by using aggregation.
+
+2. We can then use aggregation function to count the number of appearnces of a followee, by adding `count(u2) as follower_count` to the `RETURN` statement:
+```Rust
+    let result = conn.query("MATCH (u1:User)-[f:FOLLOWS]->(u2:User) RETURN u2.username, count(u2) as follower_count")?;
+    for row in result {
+        let row_str = row.iter()
+            .map(|col| col.to_string())  // Convert each column to a string
+            .collect::<Vec<String>>()    // Collect into a Vec<String>
+            .join(", ");                 // Join elements with ", "
+    
+        println!("{}", row_str);
+    }
+```
+Which returns:
```

Comment:
Can we return fewer results? using a LIMIT keyword. This is a lot of data being returned for an example. Also, why are we formatting the output this way? If the intent is to output the results to a CSV or any other downstream use, we should explain briefly what we're aiming to output from this query.